### DISCUSSION TITLE
Turn off automatic apply of Closed Caption tracks

### DISCUSSION THREAD

**Discussion by PedjaCucak** [videojs/video.js#8365] [Q&A]
Is it possible to turn off automatic apply of Closed Caption tracks (never mind that they have DEFAULT and AUTOSELECT)?

If in manifest I have something like this:

#EXT-X-MEDIA:TYPE=CLOSED-CAPTIONS,GROUP-ID="Closed Captions",NAME="en",DEFAULT=YES,AUTOSELECT=YES,LANGUAGE="en",INSTREAM-ID="CC1"

Video.js will automatic show this track on screen, is there a way to turn this off, I don't want player to show track automatic, never mind it has default and autoselect set to true?

Is there some property with which I can configure player not to show CC tracks automatically? 
Show them only when user select CC track manually.

Thanks in advance,
Pedja

**Comment by amtins** [comment]
Hi @PedjaCucak If you can modify the playlist, it is possible to set the `AUTOSELECT=NO` attribute, see HLS specification [EXT-X-MEDIA](https://datatracker.ietf.org/doc/html/draft-pantos-hls-rfc8216bis#section-4.4.6.1). Here's what's done at http-streaming level : https://github.com/videojs/http-streaming/blob/main/src/media-groups.js#L603

Alternatively, you can do the following:

If the `src` method is used

```javascript
player.getCache().selectedLanguage = { enabled: false };

player.src('https://d2zihajmogu5jn.cloudfront.net/bipbop-advanced/bipbop_16x9_variant.m3u8');
```
If the `loadMedia` method is used

```javascript
player.loadMedia(
  {
    src: 'https://d2zihajmogu5jn.cloudfront.net/bipbop-advanced/bipbop_16x9_variant.m3u8'
  },
  () => {
    player.getCache().selectedLanguage = { enabled: false };
  }
);
```

In some cases, the above solution may not work, in which case you can do the following:
```javascript
// Disables subtitles every time you load new media
player.on('loadedmetadata', ()=> {
  player.one('play', ()=> {
    Array.from(player.textTracks()).filter(({kind}) => !['chapters','metadata'].includes(kind)).forEach((track) => track.mode = "disabled")
  });
});

player.src('https://d2zihajmogu5jn.cloudfront.net/bipbop-advanced/bipbop_16x9_variant.m3u8');
// Or
player.loadMedia({
    src: 'https://d2zihajmogu5jn.cloudfront.net/bipbop-advanced/bipbop_16x9_variant.m3u8'
});
```

With all these suggestions you can choose what suits you best or it will help you find another approach.

**Reply by PedjaCucak** [reply]
Hi @amtins sure I've already saw this https://github.com/videojs/http-streaming/blob/main/src/media-groups.js#L603 but my question is, is there a way to change this track objects that I'm receiving from http-streaming, to change them in a way that cc track won't be played automatically?  I want to avoid changing http-stream source code :-)
